lib/axon/compiler.ex

```diff
@@ -113,14 +113,14 @@ defmodule Axon.Compiler do
 
     fn params, inputs ->
       inputs = maybe_flatten(inputs)
-      {fun, _} = to_predict_fun(graph, %{}, input_map)
+      {expr, _} = to_predict_fun(graph, %{}, input_map, params, inputs)
```

Comment:
As we discussed, the anonymous function will be stored in ETS, so we want to make all variables used inside the anonymous function are lightweight. So are we sure the `graph` and the `input_map` do not have large tensors in them?

Reply:
`input_map` just maps layer names to indices in the `input` from `predict_fn(params, input)` so it will never contain large tensors. It is possible that the graph contains large tensors if the user embeds them in the graph with `Axon.constant`

Reply:
Both `params` and `inputs` come from the top level function definition and will get a map and tuple of large tensors respectively - is that going to be problematic?

Reply:
Those are fine. We cache only the shapes!